src/server/debug/server/server.go

```diff
@@ -601,7 +623,96 @@ func (s *debugServer) collectPipelineDumpFunc(pachClient *client.APIClient, limi
 		if err := s.collectCommits(tw, pachClient, client.NewRepo(pipelineInfo.Pipeline.Name), limit, prefix...); err != nil {
 			return err
 		}
-		return s.collectJobs(tw, pachClient, pipelineInfo.Pipeline.Name, limit, prefix...)
+		if err := s.collectJobs(tw, pachClient, pipelineInfo.Pipeline.Name, limit, prefix...); err != nil {
+			return err
+		}
+		if err := s.forEachWorkerLoki(pipelineInfo, func(pod string) error {
+			workerPrefix := join(podPrefix, pod)
+			if len(prefix) > 0 {
+				workerPrefix = join(prefix[0], workerPrefix)
+			}
+			userPrefix := join(workerPrefix, client.PPSWorkerUserContainerName)
+			if err := s.collectLogsLoki(tw, pod, client.PPSWorkerUserContainerName, userPrefix); err != nil {
+				return err
+			}
+			sidecarPrefix := join(workerPrefix, client.PPSWorkerSidecarContainerName)
+			return s.collectLogsLoki(tw, pod, client.PPSWorkerSidecarContainerName, sidecarPrefix)
+		}); err != nil {
+			name := "loki"
+			if len(prefix) > 0 {
+				name = join(prefix[0], name)
+			}
+			return writeErrorFile(tw, err, name)
+		}
+		return nil
+	}
+}
+
+func (s *debugServer) forEachWorkerLoki(pipelineInfo *pps.PipelineInfo, cb func(string) error) error {
+	pods, err := s.getWorkerPodsLoki(pipelineInfo)
```

Comment:
am I understanding correctly that this reads through all the pipeline logs to find unique `pod` label values?

it seems like there should be some way to use the `GET /loki/api/v1/label/<name>/values` endpoint and select the ones that match the expected pod names for a pipeline, but maybe that's not worth it for now

Reply:
Yeah, I didn't see a clear way to get the values based on the value of a different label. That endpoint gives values for a particular label.